### PR TITLE
Fix stepLR + make model more flexible with arbitrary n_classes

### PR DESCRIPTION
**Pull request by hnguyentt**
Thank you for your neat code.

When I run the code with my data, I found these two issues:
* If `lr_scheduler` is `step`, there will be an error because there is no argument `gamma` for the scheduler.
* When training with 2 classes of output, the `print_progress` and `print_epoch_progress` functions in `utils.py` will result in the `tuple index out of range` error as the functions were designed for 3 classes only.

I fixed these issues. If it is not so important, just ignore the pull request.

### CODE REVIEW

**Comment by bkong999** [comment]
> Thank you for your neat code.
> 
> When I run the code with my data, I found these two issues:
> 
> * If `lr_scheduler` is `step`, there will be an error because there is no argument `gamma` for the scheduler.
> * When training with 2 classes of output, the `print_progress` and `print_epoch_progress` functions in `utils.py` will result in the `tuple index out of range` error as the functions were designed for 3 classes only.
> 
> I fixed these issues. If it is not so important, just ignore the pull request.

Thanks a lot for pointing out these issues and making contributions. Your pull requests have been merged.
